web/src/reducers/dataReducer.js

```diff
@@ -52,20 +52,22 @@ const reducer = createReducer(initialState, (builder) => {
 
       state.zoneDatetimes = { ...state.zoneDatetimes, [stateAggregation]: datetimes.map((dt) => new Date(dt)) };
       state.isLoadingGrid = false;
+      state.failedRequestType = null;
       state.hasInitializedGrid = true;
       state.isGridExpired[stateAggregation] = false;
     })
     .addCase(GRID_DATA_FETCH_REQUESTED, (state) => {
       state.isLoadingGrid = true;
-      state.hasConnectionWarning = false;
     })
     .addCase(GRID_DATA_FETCH_FAILED, (state) => {
-      state.hasConnectionWarning = true;
+      state.failedRequestType = 'grid';
```

Comment:
What do you think about storing the the names as constants, we have contants.js file where some of these things could be stored.

Eg:

failedRequestType : {
GRID: 'grid',
ZONE: 'zone',
}

And then we access it through failedRequestType.GRID